src/box/txn_limbo.h

```diff
@@ -156,6 +175,11 @@ struct txn_limbo {
 	 * limbo and raft are in sync and the terms are the same.
 	 */
 	uint64_t promote_greatest_term;
```

Comment:
Doesn't this now become "promote greatest confirmed term"? AFAIU, it gets updated only when a PROMOTE gets confirmed. Not when it is just written to WAL. Otherwise imagine this:
1. Node 1 write PROMOTE and hangs.
2. Node 2 bumps term a few times and write PROMOTE too.
3. Now Node-2-PROMOTE gets delivered to Node-1. It will pass all filters until `txn_limbo_filter_promote()`. Here `req->prev_term == limbo->promote_greatest_term` would be `false` and `rlist_foreach_entry_reverse()` won't find the preceeding PROMOTE. Because Node-1 has PROMOTE say term 10 and Node-2's PROMOTE can be 20.

But it seems to work fine when `promote_greatest_term` is updated only once a new PROMOTE gets confirmed. And that looks so in the code too.

Did I get it right? Should we also update this comment then? Right now it says that the term is updated right after PROMOTE WAL write.